File: testdata/ts_small/src/db/queries.ts
/** Database query helpers. */

import { getConnection } from "./connection";
import type { User } from "../types";

export function findUserById(id: string): User | null {
  const conn = getConnection();
  return conn.query("SELECT * FROM users WHERE id = ?", [id]);
}

export function findUserByEmail(email: string): User | null {
  const conn = getConnection();
  return conn.query("SELECT * FROM users WHERE email = ?", [email]);
}

export function insertUser(user: User): void {
  const conn = getConnection();
  conn.execute("INSERT INTO users VALUES (?, ?, ?, ?)", [
    user.id,
    user.username,
    user.email,
    user.role,
  ]);
}

File: testdata/python_small/mylib/db/queries.py
"""Database query functions."""

from mylib.db.connection import get_connection
from mylib.models import User


def find_user(username):
    """Find a user by username."""
    conn = get_connection()
    row = conn.get("SELECT * FROM users WHERE username = ?", (username,))
    if row:
        return User(row["username"], row["email"])
    return None


def find_all_users():
    """Return all users."""
    conn = get_connection()
    rows = conn.get("SELECT * FROM users")
    return [User(r["username"], r["email"]) for r in rows]


def delete_user(username):
    """Delete a user by username."""
    conn = get_connection()
    conn.execute("DELETE FROM users WHERE username = ?", (username,))
